### PR TITLE
SSCS-10152 Security, dependency and general fixes.

### PR DESCRIPTION
**Pull request by dattipoeHMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/sscs/SecurityConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            .build();
+
+        given(requestSpecification)
+            .when()
+            .get("/hearing")
+            .then()
+            .assertThat()
+            .statusCode(HttpStatus.UNAUTHORIZED.value());
+    }
+
```
Comment (dattipoeHMCTS) [inline src/functionalTest/java/uk/gov/hmcts/reform/sscs/SecurityConfigurationTest.java:38]: Positive case will be added once we have an actual endpoint.

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -193,7 +193,7 @@
     <module name="CustomImportOrder">
       <property name="sortImportsInGroupAlphabetically" value="true"/>
       <property name="separateLineBetweenGroups" value="true"/>
-      <property name="customImportOrderRules" value="THIRD_PARTY_PACKAGE###STANDARD_JAVA_PACKAGE###STATIC"/>
+      <property name="customImportOrderRules" value="STATIC###THIRD_PARTY_PACKAGE"/>
```
Comment (TeaWithLucas) [inline config/checkstyle/checkstyle.xml:196]: That is the HMCTS standard, most projects use it, what's the reason for the change?
https://github.com/search?q=org%3Ahmcts+customImportOrderRules&type=code

File: build.gradle [hunk 3]
```diff
@@ -204,117 +216,83 @@ repositories {
...
-    }
-    implementation(group: 'org.bouncycastle', name: 'bcprov-jdk15on', version: '1.64') {
-      because 'CVE-2020-15522'
-    }
-    implementation(group: 'com.thoughtworks.xstream', name: 'xstream', version: ' 1.4.19') {
-      because 'CVE-2021-43859'
-    }
+dependencyManagement {
+  imports {
+    mavenBom "org.springframework.cloud:spring-cloud-dependencies:${versions.springCloud}"
```
Comment (TeaWithLucas) [inline build.gradle:298]: What is the benefit/difference between using dependencyManagement and constraints?

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -21,54 +21,31 @@ management:
...
-#      poolName: {to-be-defined}HikariCP
-#      maxLifetime: 7200000
-#      connectionTimeout: 30000
-#  jpa:
-#    properties:
-#      hibernate:
-#        jdbc:
-#          lob:
-#            # silence the 'wall-of-text' - unnecessary exception throw about blob types
-#            non_contextual_creation: true
```
Comment (TeaWithLucas) [inline src/main/resources/application.yaml:71]: Will we not be using a database for the event queue?

File: build.gradle (hunk 3 above)
Comment (dattipoeHMCTS) [reply build.gradle:308]: Oops, I need to check my Github email rule.

I'm only using Spring's dependency management here to control what versions of Spring Cloud get resolved. If you noticed I haven't provided an explicit 3.1.0 version for spring-cloud-starter. The BOM controls what versions get put in.

I was going to use constraints to fix transitive dependency versions but John suggested that we should suppress them instead to reduce the chance of any incompatibilities arising.

File: config/checkstyle/checkstyle.xml (hunk 2 above)
Comment (dattipoeHMCTS) [reply config/checkstyle/checkstyle.xml:196]: I had such a fight with Checkstyle and IntelliJ. I changed it to what tribunals-api uses, if it's standard across I will change it back.

File: build.gradle [hunk 5]
```diff
@@ -204,117 +233,90 @@ repositories {
...
-
-ext.libraries = [
-  junit5: [
-    "org.junit.jupiter:junit-jupiter-api:${versions.junit}",
-    "org.junit.jupiter:junit-jupiter-engine:${versions.junit}",
-    "org.junit.jupiter:junit-jupiter-params:${versions.junit}",
-    "org.junit.platform:junit-platform-commons:${versions.junitPlatform}",
-    "org.junit.platform:junit-platform-engine:${versions.junitPlatform}"
-  ]
+  springCloud: '2021.0.1',
```
Comment (johnoneill46) [inline build.gradle:239]: don't think this is right?

File: Jenkinsfile_CNP [hunk 6]
```diff
@@ -1,40 +1,50 @@
 #!groovy
-
 @Library("Infrastructure")
+import java.lang.Object
 
 def type = "java"
 def product = "sscs"
 def component = "hearings-api"
+def branchesToSync = ['demo', 'ithc', 'perftest']
```
Comment (johnoneill46) [inline Jenkinsfile_CNP:8]: where's this used?

File: build.gradle [hunk 7]
```diff
@@ -204,117 +233,90 @@ repositories {
...
+  restAssured: '4.5.1',
+  log4J: '2.17.1',
+  assertJ: '3.22.0',
+  lombok: '1.18.22',
+  springRetry: '1.3.1',
+  serviceAuthProviderJavaClient: '4.0.3',
+  gradlePiTestPlugin: '1.5.2',
+  sonarPiTestPlugin: '0.5',
+  pactJunit: '4.3.5',
+  pactJava8: '4.1.34'
```
Comment (johnoneill46) [inline build.gradle:253]: Keep these in alphabetical order

**Review by johnoneill46 — CHANGES_REQUESTED** [review]

File: Jenkinsfile_CNP (hunk 6 above)
Comment (dattipoeHMCTS) [reply Jenkinsfile_CNP:8]: Down on line 40

File: build.gradle (hunk 5 above)
Comment (dattipoeHMCTS) [reply build.gradle:239]: The spring cloud version? They have a weird naming convention but this is the BOM version: https://spring.io/blog/2022/02/18/spring-cloud-2021-0-1-has-been-released

**Review by johnoneill46 — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

**Review by ss-evoco — APPROVED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Pull request merged**